packages/MinimalScope/tests/Rector/Class_/ChangeLocalPropertyToVariableRector/Fixture/fixture.php.inc

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Rector\MinimalScope\Tests\Rector\Class_\ChangeLocalPropertyToVariableRector\Fixture;
+
+class SomeClass
+{
+    private $count;
+    public function run()
```

Comment:
Wouldnt the rector fail when the method e.g. looks like

```
public function run()
     {
         If ($this->count == 5) {
           $this->count = 10;
           return $this->count;
         }
         $this->count = 5;
         return $this->count;
     }
```

Reply:
Why should it fail?

Reply:
I guess the logic will change because of the rector.

In case it gets transformed into
```

public function run()
     {
         If ($count == 5) {
           $count = 10;
           return $count;
         }
         $count = 5;
         return $count;
     }
```

The method will return different values when invoked 2 times in a row

Reply:
> The method will return different values when invoked 2 times in a row

I see. That's what I was missing.

How would you approach it?

Reply:
I guess it could work when you made sure that there is only a single property-write

Reply:
That sound clear, I'll add fixture for that. Thanks for spotting this :+1:

Reply:
https://github.com/rectorphp/rector/pull/2563#merged-event

Reply:
It might needs full new service to analyze scope in the future. Let's see how it handles reality now